core/src/main/java/org/apache/calcite/prepare/CalciteCatalogReader.java

```diff
@@ -97,6 +97,34 @@ public CalciteCatalogReader(CalciteSchema rootSchema,
         typeFactory, config);
   }
 
+
+  public CalciteCatalogReader(CalciteSchema rootSchema,
+      List<List<String>> defaultSchemas,
+      int numDefaults, // This field isn't needed but ensure the signatures are unique.
```

Comment:
So this API is different because of weird Java quirk. List<String> and List<List<String>> both map to the same generic, so just replacing the schema results in a duplicate API. This extra field makes them different an enables some validation in testing.